syntax = "proto3";

package proto_serialization;

import "map_renderer.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string name = 1;
    Coordinates coords = 2;
}

message Bus {
    string name = 1;
    repeated Stop stops = 2;
    bool is_circular = 3;
}

message Distance {
    string from = 1;
    string to = 2;
    uint32 distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
    RenderSettings render_settings = 4;
}
